### PR TITLE
[DOC] RDoc for process.c

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: process.c [hunk 1]
```diff
@@ -5123,12 +5120,13 @@ proc_setpgrp(VALUE _)
 #if defined(HAVE_GETPGID)
 /*
  *  call-seq:
- *     Process.getpgid(pid)   -> integer
+ *    Process.getpgid(pid) -> integer
  *
- *  Returns the process group ID for the given process id. Not
- *  available on all platforms.
+ *  Returns the process group ID for the given process id +pid+:
```
Comment (peterzhu2118) [inline process.c:5125]: Nit: capitalize ID.

```suggestion
 *  Returns the process group ID for the given process ID +pid+:
```

File: process.c [hunk 2]
```diff
@@ -5173,15 +5173,16 @@ proc_setpgid(VALUE obj, VALUE pid, VALUE pgrp)
 #ifdef HAVE_GETSID
 /*
  *  call-seq:
- *     Process.getsid()      -> integer
- *     Process.getsid(pid)   -> integer
+ *    Process.getsid(pid = nil) -> integer
+ *
+ *  Returns the session ID of the given process id +pid+,
```
Comment (peterzhu2118) [inline process.c:5178]:
```suggestion
 *  Returns the session ID of the given process ID +pid+,
```

File: process.c [hunk 3]
```diff
@@ -5208,13 +5209,15 @@ static rb_pid_t ruby_setsid(void);
...
  *  call-seq:
- *     Process.setsid   -> integer
+ *    Process.setsid -> integer
  *
- *  Establishes this process as a new session and process group
- *  leader, with no controlling tty. Returns the session id. Not
- *  available on all platforms.
+ *  Establishes the current process as a new session and process group leader,
+ *  with no controlling tty;
+ *  returns the session id:
```
Comment (peterzhu2118) [inline process.c:5216]:
```suggestion
 *  returns the session ID:
```

File: process.c [hunk 4]
```diff
@@ -5262,19 +5265,26 @@ ruby_setsid(void)
...
+ *  or user.
+ *
+ *  Argument +kind+ is one of:
+ *
+ *  - Process::PRIO_PROCESS: return priority for process.
+ *  - Process::PRIO_PGRP: return priority for process group.
+ *  - Process::PRIO_USER: return priority for user.
+ *
+ *  Argument +id+ is the ID for the process, process group, or user;
+ *  zero specified the current id for +kind+.
```
Comment (peterzhu2118) [inline process.c:5280]:
```suggestion
 *  zero specified the current ID for +kind+.
```

File: process.c [hunk 5]
```diff
@@ -5553,22 +5567,16 @@ rlimit_resource_value(VALUE rval)
...
- *  _resource_ indicates the kind of resource to limit.
- *  It is specified as a symbol such as <code>:CORE</code>,
- *  a string such as <code>"CORE"</code> or
- *  a constant such as Process::RLIMIT_CORE.
- *  See Process.setrlimit for details.
- *
- *  _cur_limit_ and _max_limit_ may be Process::RLIM_INFINITY,
- *  Process::RLIM_SAVED_MAX or
- *  Process::RLIM_SAVED_CUR.
- *  See Process.setrlimit and the system getrlimit(2) manual for details.
```
Comment (peterzhu2118) [inline process.c:5571]: I think this part of the documentation is important. We should keep some of it.

File: process.c [hunk 6]
```diff
@@ -5588,54 +5596,54 @@ proc_getrlimit(VALUE obj, VALUE resource)
...
+ *
+ *  Argument +resource+ specifies the resource whose limits are to be set;
+ *  the argument may be given as a symbol, as a string, or as a constant
+ *  beginning with <tt>Process::RLIMIT_</tt>
+ *  (e.g., +:CORE+, <tt>'CORE'</tt>, or <tt>Process::RLIMIT_CORE</tt>.
+ *
+ *  The resources available and supported are system-dependent,
+ *  and may include (here expressed as symbols):
+ *
+ *  - +:AS+: Total available memory (bytes) (SUSv3, NetBSD, FreeBSD, OpenBSD but 4.4BSD-Lite).
```
Comment (peterzhu2118) [inline process.c:5613]: I think the `but` here actually means `except`.

```suggestion
 *  - +:AS+: Total available memory (bytes) (SUSv3, NetBSD, FreeBSD, OpenBSD except 4.4BSD-Lite).
```

File: process.c (hunk 5 above)
Comment (BurdetteLamar) [reply process.c:5571]: Revised.  I'm not sure how these constants work with the two methods.

**Comment by BurdetteLamar** [comment]
Some of these methods could probably profit by having  links to the outboard system docs, and maybe to Wikipedia articles;  could be in next PR?

**Comment by peterzhu2118** [comment]
@BurdetteLamar I don't see most of these comments as being addressed. Did you miss a commit?

**Comment by BurdetteLamar** [comment]
> @BurdetteLamar I don't see most of these comments as being addressed. Did you miss a commit?

I pushed the one revision, then tried to commit your suggestions;  missed the message that the diff was outdated.  Will try again.

File: process.c [hunk 7]
```diff
@@ -5588,54 +5604,54 @@ proc_getrlimit(VALUE obj, VALUE resource)
...
+ *
+ *  Arguments +cur_limit+ and +max_limit+ may be:
+ *
+ *  - Integers (+max_limit+ should not be smaller than +cur_limit+).
+ *  - Symbol +:SAVED_MAX+, string <tt>'SAVED_MAX'</tt>,
+ *    or constant <tt>Process::RLIM_SAVED_MAX: saved maximum limit.
+ *  - Symbol +:SAVED_CUR+, string <tt>'SAVED_CUR'</tt>,
+ *    or constant <tt>Process::RLIM_SAVED_CUR: saved current limit.
+ *  - Symbol +:INFINITY+, string <tt>'INFINITY'</tt>,
+ *    or constant <tt>Process::RLIM_INFINITY: no limit on resource.
```
Comment (peterzhu2118) [inline process.c:5647]: I think you're missing the closing `</tt>` tag here.

```suggestion
 *  - Integers (+max_limit+ should not be smaller than +cur_limit+).
 *  - Symbol +:SAVED_MAX+, string <tt>'SAVED_MAX'</tt>,
 *    or constant <tt>Process::RLIM_SAVED_MAX</tt>: saved maximum limit.
 *  - Symbol +:SAVED_CUR+, string <tt>'SAVED_CUR'</tt>,
 *    or constant <tt>Process::RLIM_SAVED_CUR</tt>: saved current limit.
 *  - Symbol +:INFINITY+, string <tt>'INFINITY'</tt>,
 *    or constant <tt>Process::RLIM_INFINITY</tt>: no limit on resource.
```

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**